src/wallet/wallet2.cpp

```diff
@@ -2131,6 +2144,12 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
               error::wallet_internal_error, "Unexpected values of new and old outputs");
           tx_money_got_in_outs[tx_scan_info[o].received->index] -= m_transfers[kit->second].amount();
 
+          amounts_container& tx_amounts_this_out = tx_amounts_individual_outs[tx_scan_info[o].received->index]; // Only for readability on the following lines
+          auto amount_iterator = std::find(tx_amounts_this_out.begin(), tx_amounts_this_out.end(), m_transfers[kit->second].amount());
+          THROW_WALLET_EXCEPTION_IF(amount_iterator == tx_amounts_this_out.end(),
+              error::wallet_internal_error, "Unexpected values of new and old outputs");
+          tx_amounts_this_out.erase(amount_iterator);
```

Comment:
This one will fail to add the difference amount, which might be better. Admittedly a corner case due to the fact duplicate pubkeys are allowed.

Reply:
I admit that I don't follow. 

In this case, we have a larger output "replacing" a smaller one with the same pubkey. A couple of lines above, we fix the value of `tx_money_got_in_outs[tx_scan_info[o].received->index]` because of this - because both outputs' amounts have been added. 

If both outputs' (with the same pubkey) amounts have been added to `tx_money_got_in_outs`, they should have been added to the new variable `tx_amounts_individual_outs` as well. Correct? 

> Admittedly a corner case due to the fact duplicate pubkeys are allowed.

This is precisely the case of duplicate pubkeys. `kit !=  m_pub_keys.end()` and `m_transfers[kit->second].amount() <= tx_scan_info[o].amount` in this branch.

> This one will fail to add the difference amount, which might be better.

What difference amount do you mean? What "might be better"?

Thanks :)

Reply:
If the new output is 1, replacing one with .2, you remove the 1, but don't seem to add .8 instead.

Reply:
And "might be better", I meant I'm not sure that adding .8 is that much better, since it does not match the actual output. Keeping 1 might be better. But nothing at all seems objectively worse.

Reply:
Got it, makes sense. Removing this whole piece of code to keep the actual output. Thanks for the clarification.